server/services/prisonerProfileService.ts

```diff
@@ -152,31 +187,55 @@ export default class PrisonerProfileService {
 
     const visitsForDisplay: PastVisitItem[] = await Promise.all(
       socialVisits.map(async visit => {
-        const visitors: number[] = visit.visitors.reduce((personIds, visitor) => {
-          personIds.push(visitor.nomisPersonId)
-
-          return personIds
-        }, [])
-        const visitContactNames = await this.getPrisonerSocialContacts(contacts, visitors)
+        const visitContactNames = this.getPrisonerSocialContacts(contacts, visit.visitors)
 
         return [
-          { html: `${properCase(visit.visitType)}<br>(${properCase(visit.visitRestriction)})` },
-          { text: 'Hewell (HMP)' },
+          {
+            html: formatVisitType({ visitType: visit.visitType, visitRestriction: visit.visitRestriction }),
+            attributes: {
+              'data-test': 'tab-past-type',
+            },
+          },
+          {
+            text: 'Hewell (HMP)',
+            attributes: {
+              'data-test': 'tab-past-location',
+            },
+          },
           {
             html: visit.startTimestamp
               ? `<p>${visitDateAndTime({ startTimestamp: visit.startTimestamp, endTimestamp: visit.endTimestamp })}</p>`
               : '<p>N/A</p>',
+            attributes: {
+              'data-test': 'tab-past-date-and-time',
+            },
+          },
+          {
+            html: `<p>${visitContactNames.join('<br>')}</p>`,
+            attributes: {
+              'data-test': 'tab-past-visitors',
+            },
+          },
+          {
+            text: `${properCase(visit.visitStatus)}`,
+            attributes: {
+              'data-test': 'tab-past-status',
+            },
           },
-          { html: `<p>${visitContactNames.join('<br>')}</p>` },
-          { text: `${properCase(visit.visitStatus)}` },
         ] as PastVisitItem
       })
     )
 
     return visitsForDisplay
   }
 
-  private async getPrisonerSocialContacts(contacts: Contact[], contactIds: number[]): Promise<string[]> {
+  private getPrisonerSocialContacts(contacts: Contact[], visitors: Visitor[]): string[] {
```

Comment:
Not strictly needed but spotted it whilst doing other bits, removes more duplication.

Reply:
Also realised it wasn't actually an async function.